### PR TITLE
Customize style text view

### PR DESCRIPTION
**Pull request by Sulfkain**
I create a Builder to set the style to apply on the TextView.

I have also created some test which I think covers all cases.
PR requested as commented [here](https://github.com/burhanrashid52/PhotoEditor/issues/142)

### CODE REVIEW

File: photoeditor/src/main/java/ja/burhanrashid52/photoeditor/TextStyleBuilder.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+import android.graphics.Typeface;
+import android.graphics.drawable.Drawable;
+import android.os.Build;
+import android.support.annotation.NonNull;
+import android.widget.TextView;
+
+import java.util.HashMap;
+import java.util.Map;
+
```
Comment (burhanrashid52) [inline photoeditor/src/main/java/ja/burhanrashid52/photoeditor/TextStyleBuilder.java]: Need to add java doc for public API which our `TextStyleBuilder` class

File: photoeditor/src/main/java/ja/burhanrashid52/photoeditor/TextStyleBuilder.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    protected void applyTextAppearance(TextView textView, int styleAppearance) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            textView.setTextAppearance(styleAppearance);
+        } else {
+            textView.setTextAppearance(textView.getContext(), styleAppearance);
+        }
+    }
+
+    protected enum TextStyle {
+        SIZE("TextSize"),
```
Comment (burhanrashid52) [inline photoeditor/src/main/java/ja/burhanrashid52/photoeditor/TextStyleBuilder.java]: Should go with TextStyle properties values to all small or all caps

File: photoeditor/src/main/java/ja/burhanrashid52/photoeditor/TextStyleBuilder.java (hunk 1 above)
Comment (Sulfkain) [reply photoeditor/src/main/java/ja/burhanrashid52/photoeditor/TextStyleBuilder.java]: Ok, but I will add it, but look that it's private on lib, no client could use this by his own

File: photoeditor/src/main/java/ja/burhanrashid52/photoeditor/TextStyleBuilder.java (hunk 2 above)
Comment (Sulfkain) [reply photoeditor/src/main/java/ja/burhanrashid52/photoeditor/TextStyleBuilder.java]: I don't get that, you mean the name of enum? if so, no, it should be like that  => [official doc](https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html)

If you mean the value, the properties don't need any format, it's a value wich could be any as you want, and I always prefer to write the values with camelcase

**Comment by burhanrashid52** [comment]
Just one more thing. To demonstrate the API, is it possible for you to add this text builder usage in the sample?

**Comment by Sulfkain** [comment]
Of course, I have in my local git, I upload it, but the UI is just prepared to change the color, so if I change the size or any other param will be weird.
So the example will be pretty simple with actual style.

**Comment by Sulfkain** [comment]
You know where will be a release with this?

**Comment by burhanrashid52** [comment]
The release is ready. But I am adding test cases as much as I could in #146 .After completing that I'll merge that in release and will release 0.4.0 version with your changes. You can also help by adding tests for the library. It will help to speed up the release @Sulfkain

**Pull request merged**